src/sentry/api/base.py

```diff
@@ -213,14 +218,31 @@ def permission_denied(self, request, message=None, code=None):
         the appropriate exception according to parent DRF function.
         """
         permissions = self.get_permissions()
+        if request.user.is_authenticated and len(permissions) == 1:
+            permission_cls = permissions[0]
+            enforce_staff_permission = features.has(
+                "auth:enterprise-staff-cookie", actor=request.user
+            )
 
-        can_be_superuser = request.user.is_authenticated and request.user.is_superuser
-        has_only_superuser_permission = len(permissions) == 1 and isinstance(
-            permissions[0], SuperuserPermission
-        )
+            # TODO(schew2381): Remove SuperuserOrStaffFeatureFlaggedPermission
+            # from isinstance checks once feature flag is removed.
+            if enforce_staff_permission:
+                is_staff_user = request.user.is_staff
+                has_only_staff_permission = isinstance(
+                    permission_cls, (StaffPermission, SuperuserOrStaffFeatureFlaggedPermission)
+                )
+
+                if enforce_staff_permission and is_staff_user and has_only_staff_permission:
```

Comment:
```suggestion
                if is_staff_user and has_only_staff_permission:
```
nit: i think `enforce_staff_permission` is duplicated here